Content.Server/Nyanotrasen/Mail/MailSystem.cs

```diff
@@ -105,19 +95,20 @@ public override void Initialize()
         public override void Update(float frameTime)
         {
             base.Update(frameTime);
-            foreach (var mailTeleporter in EntityQuery<MailTeleporterComponent>())
+
+            var query = EntityQueryEnumerator<MailTeleporterComponent, ApcPowerReceiverComponent>();
+            while (query.MoveNext(out var uid, out var mailTeleporter, out var power))
             {
-                if (TryComp<ApcPowerReceiverComponent>(mailTeleporter.Owner, out var power) && !power.Powered)
-                    return;
+                if (!power.Powered)
+                    continue;
```

Comment:
An unpowered mail teleporter should still be able to work - I think the old version here (_using continue rather than return holy shit_) is better.

Reply:
I don't think I get it..? Mail teleporters *should* require power to work, so now if the teleporter is unpowered, it skips that iteration but keeps processing other teleporters.

Reply:
Imagine you have an entity that should function as a mail teleporter without having an ApcPowerReceiverComponent (maybe some admeme teleporter on a mail ship that FTLs in).  That is skipped by this.  The previous implementation checks "if this needs power but doesn't have it, stop everything".  The check makes sense, but you should still move on to the next mail teleporter.

Reply:
OH yeah I get it.

Reply:
Thanks, sorry to pester.